File: 240101/격자 위의 편안한 상태/comfortable-state-on-the-grid.py
n, m = tuple(map(int, input().split()))
grid = [
    [0] * n
    for _ in range(n)
]
checked = [
    [False] * n
    for _ in range(n)
]

querys = [
    tuple(map(int, input().split()))
    for _ in range(m)
]

def in_range(x, y):
    return 0 <= x and x < n and 0 <= y and y < n


def comfortable(x, y):
    return in_range(x, y) and grid[x][y] == 1
    

dxs, dys = [-1, 1, 0, 0], [0, 0, -1, 1]
for r, c in querys:
    x, y = r-1, c-1
    grid[x][y] = 1
    cnt = 0
    
    for dx, dy in zip(dxs, dys):
        nx, ny = x + dx, y + dy
        if comfortable(nx, ny):
            cnt += 1
    
    if cnt == 3:
        print(1)
    else:
        print(0)

File: 240101/격자 위의 편안한 상태/comfortable-state-on-the-grid.java
import java.util.Scanner;

public class Main {
    public static final int MAX_N = 100;
    public static int n, m, x, y;
    public static int[] dxs = new int[]{-1, 1, 0, 0};
    public static int[] dys = new int[]{0, 0, -1, 1};
    public static int[][] grid = new int[MAX_N][MAX_N];
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        m = sc.nextInt();

        for (int i = 0; i < m; i++) {
            int r = sc.nextInt();
            int c = sc.nextInt();
            x = r-1; y = c-1;
            
            grid[x][y] = 1;

            if(isComfort(x, y))
                System.out.println(1);
            else
                System.out.println(0);
        }
        
        
    }

    private static boolean isComfort(int x, int y) {
        int cnt = 0;

        for (int i = 0; i < 4; i++) {
            int nx = x + dxs[i], ny = y + dys[i];

            if(inRange(nx, ny) && grid[nx][ny] == 1)
                cnt++;
        }
        
        return cnt == 3;
    }

    private static boolean inRange(int x, int y) {
        return 0 <= x && x < n && 0 <= y && y < n;
    }
    
}